src/serialization.jl

```diff
@@ -1,15 +1,76 @@
 """
-    restoreoptsum!(m::LinearMixedModel, io::IO; atol::Real=0, rtol::Real=atol>0 ? 0 : √eps)
-    restoreoptsum!(m::LinearMixedModel, filename; atol::Real=0, rtol::Real=atol>0 ? 0 : √eps)
+    restoreoptsum!(m::MixedModel, io::IO; atol::Real=0, rtol::Real=atol>0 ? 0 : √eps)
+    restoreoptsum!(m::MixedModel, filename; atol::Real=0, rtol::Real=atol>0 ? 0 : √eps)
 
 Read, check, and restore the `optsum` field from a JSON stream or filename.
 """
-function restoreoptsum!(
-    m::LinearMixedModel{T}, io::IO; atol::Real=zero(T),
-    rtol::Real=atol > 0 ? zero(T) : √eps(T),
-) where {T}
+function restoreoptsum!(m::MixedModel, filename; kwargs...)
+    return open(filename, "r") do io
+        return restoreoptsum!(m, io; kwargs...)
+    end
+end
+
+function restoreoptsum!(m::LinearMixedModel{T}, io::IO;
+    atol::Real=zero(T),
+    rtol::Real=atol > 0 ? zero(T) : √eps(T)) where {T}
+    dict = JSON3.read(io)
+    ops = restoreoptsum!(m.optsum, dict)
+    for (par, obj_at_par) in (:initial => :finitial, :final => :fmin)
+        if !isapprox(
+            objective(updateL!(setθ!(m, getfield(ops, par)))), getfield(ops, obj_at_par);
+            rtol, atol,
+        )
+            throw(
+                ArgumentError(
+                    "model m at $par does not give stored $obj_at_par within given tolerances"
+                ),
+            )
+        end
+    end
+    return m
+end
+
+function restoreoptsum!(m::GeneralizedLinearMixedModel{T}, io::IO;
+    atol::Real=zero(T),
+    rtol::Real=atol > 0 ? zero(T) : √eps(T)) where {T}
     dict = JSON3.read(io)
     ops = m.optsum
+
+    # need to accommodate fast and slow fits
+    resize!(ops.initial, length(dict.initial))
+    resize!(ops.final, length(dict.final))
+
+    theta_beta_len = length(m.θ) + length(m.β)
+    if length(dict.initial) == theta_beta_len # fast=false
+        if length(ops.lowerbd) == length(m.θ)
+            prepend!(ops.lowerbd, fill(-Inf, length(m.β)))
+        end
+        setpar! = setβθ!
+        varyβ = false
+    else # fast=true
+        setpar! = setθ!
+        varyβ = true
+        if length(ops.lowerbd) != length(m.θ)
+            deleteat!(ops.lowerbd, 1:length(m.β))
+        end
+    end
+    restoreoptsum!(ops, dict)
+    for (par, obj_at_par) in (:initial => :finitial, :final => :fmin)
+        if !isapprox(
+            deviance(pirls!(setpar!(m, getfield(ops, par)), varyβ), dict.nAGQ),
+            getfield(ops, obj_at_par); rtol, atol,
+        )
+            throw(
+                ArgumentError(
+                    "model m at $par does not give stored $obj_at_par within given tolerances"
```

Comment:
Same suggestion as above
```suggestion
                    "model at $par does not match stored $obj_at_par within atol=$atol, rtol=$rtol"
```